pandas/_libs/index.pyx

```diff
@@ -544,6 +544,10 @@ cpdef convert_scalar(ndarray arr, object value):
             pass
         elif isinstance(value, timedelta):
             return Timedelta(value).value
+        elif util.is_datetime64_object(value):
```

Comment:
how does this not hit the if at line 531?

Reply:
531 is a check on the `arr` variable.  This is a check on `value`.